src/test/java/org/opendevstack/provision/services/MailAdapterTest.java

```diff
@@ -14,46 +14,41 @@
 
 package org.opendevstack.provision.services;
 
+import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.mockito.ArgumentMatchers.any;
 
 import javax.mail.internet.MimeMessage;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.Mockito;
-import org.mockito.MockitoAnnotations;
 import org.opendevstack.provision.authentication.crowd.CrowdAuthenticationManager;
 import org.opendevstack.provision.model.OpenProjectData;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.mail.javamail.JavaMailSender;
-import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.context.ActiveProfiles;
 
 @SpringBootTest
-@DirtiesContext
 @ActiveProfiles("crowd")
 public class MailAdapterTest {
-  @Mock JavaMailSender mailSender;
 
-  @InjectMocks @Autowired MailAdapter mailAdapter;
+  @Autowired private MailAdapter mailAdapter;
 
-  @Autowired CrowdAuthenticationManager manager;
+  @Mock private CrowdAuthenticationManager crowdAuthenticationManager;
```

Comment:
as discussed, please replace `@Mock` in favour of `@MockBean`

Reply:
I see, the last change removed the spring test annotation for this test. Since `MailAdapter` is annotated with `@Service` and for that reason is a spring service is wiser to test it with the spring test annotation to benefit and test the DI configuration. 
However, we are having issues in getting the correct spring test configuration, which is delaying us from merging this refactoring. 
Since this refactoring improve a lot the overall tests and as I do not want to delay the merge, I propose to create 2 new issues. One to turn back the configuration of MailAdapterTest to use spring annotations. 2nd one to improve the spring test configuration which will give us performance improvements. Sound good?

Reply:
Could you explain why you think it is wiser to test services with the spring test annotation? In my opinion, in services we often only want to unit test the business logic in isolation here, which can be independent of the Spring Framework. 
I see the use case for the spring application context rather when testing the integration of multiple classes or for example the web layer in isolation with a mocked servlet environment.

Reply:
the additionally benefit of testing with spring test configuration is that the spring configuration logic and how the service is built up is also tested. Otherwise that logic is not tested.
While this goes beyond of the scope of a "unit test", it has the advantage that is a bit closer test scenario to how the service is built up later in production. Make sense?

Reply:
hm I'm not quite sure about this, even the spring doc mentions in [its part about unit testing](https://docs.spring.io/spring-framework/docs/current/reference/html/testing.html#unit-testing):

> Dependency injection should make your code less dependent on the container than it would be with traditional Java EE development. The POJOs that make up your application should be testable in JUnit or TestNG tests, with objects instantiated by using the new operator, without Spring or any other container. You can use mock objects (in conjunction with other valuable testing techniques) to test your code in isolation. If you follow the architecture recommendations for Spring, the resulting clean layering and componentization of your codebase facilitate easier unit testing. **For example, you can test service layer objects by stubbing or mocking DAO or repository interfaces**, without needing to access persistent data while running unit tests.

Testing the spring context is something they mention in the [part about integration test](https://docs.spring.io/spring-framework/docs/current/reference/html/testing.html#integration-testing-overview):

> It is important to be able to perform some integration testing without requiring deployment to your application server or connecting to other enterprise infrastructure. Doing so lets you test things such as:
> 
> -   **The correct wiring of your Spring IoC container contexts.**
> ...

but anyway, I'll put it this test class back to a state where it worked and we can fix this situation later on.